src/internal/m365/collection/exchange/restore.go

```diff
@@ -222,32 +232,76 @@ func uploadAttachments(
 ) error {
 	el := errs.Local()
 
-	for _, a := range as {
+	// Manual testing showed that even small delays between retries could get us
+	// past the issues. Use an exponential backoff that will allow us to keep
+	// runtimes low if there's just one failure for a given attachment but still
+	// waits a reasonable amount of time if there's multiple failures.
+	retryBackoff := backoff.NewExponentialBackOff()
+	retryBackoff.InitialInterval = retryInterval
+	retryBackoff.Multiplier = retryMultiplier
+	retryBackoff.MaxInterval = retryMaxInterval
+	retryBackoff.MaxElapsedTime = retryMaxElapsedTime
+
+	for i, a := range as {
 		if el.Failure() != nil {
 			return el.Failure()
 		}
 
-		err := uploadAttachment(
-			ctx,
-			ap,
-			resourceID,
-			destinationID,
-			itemID,
-			a)
-		if err != nil {
-			// FIXME: I don't know why we're swallowing this error case.
-			// It needs investigation: https://github.com/alcionai/corso/issues/3498
-			if ptr.Val(a.GetOdataType()) == "#microsoft.graph.itemAttachment" {
-				name := ptr.Val(a.GetName())
+		actx := clues.Add(ctx, "attachment_num", i)
 
-				logger.CtxErr(ctx, err).
-					With("attachment_name", name).
-					Info("mail upload failed")
+		var (
+			retry int
+			err   error
+		)
 
-				continue
+		retryBackoff.Reset()
+
+		for (retry == 0 || err != nil) && retry <= maxRetries {
+			retry++
+
+			ictx := clues.Add(actx, "attempt_num", retry)
+
+			err = uploadAttachment(
+				ictx,
+				ap,
+				resourceID,
+				destinationID,
+				itemID,
+				a)
+			if err != nil {
+				// Sometimes graph returns a 404 when we try to post the attachment.
+				// We're not sure why, but maybe it has to do with attaching many items.
+				// In any case, wait a little while and try again.
+				if graph.IsErrItemNotFound(err) {
+					if retry <= maxRetries {
+						waitTime := retryBackoff.NextBackOff()
+
+						logger.Ctx(ictx).Infow("error uploading attachment, retrying", "retry_after", waitTime)
+
+						time.Sleep(waitTime)
+					}
+
+					continue
+				}
+
```

Comment:
Nit: this doesn't need to be double nested
```suggestion
			// Sometimes graph returns a 404 when we try to post the attachment.
			// We're not sure why, but maybe it has to do with attaching many items.
			// In any case, wait a little while and try again.
			if graph.IsErrItemNotFound(err) {
				if retry <= maxRetries {
					waitTime := retryBackoff.NextBackOff()

					logger.Ctx(ictx).Infow("error uploading attachment, retrying", "retry_after", waitTime)

					time.Sleep(waitTime)
				}

				continue
			}

			if err != nil && ptr.Val(a.GetOdataType()) == "#microsoft.graph.itemAttachment"{
```